
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.4">
    
    
      
        <title>Mining EMG with R part 2A - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-769c285a91.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-02c2a4388f.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Metagenomics Modules
                </span>
              
                <span class="md-header-nav__parent">
                  Metagenomics mining EMG with R
                </span>
              
            
            Mining EMG with R part 2A
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../timetables/timetable_metagenomics/" title="Timetable" class="md-tabs__link">
          Timetable
        </a>
      
    </li>
  

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-cli/commandline/" title="Metagenomics Modules" class="md-tabs__link md-tabs__link--active">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Timetable
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Timetable
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-4" type="checkbox" id="nav-5-4" checked>
    
    <label class="md-nav__link" for="nav-5-4">
      Metagenomics mining EMG with R
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-4">
        Metagenomics mining EMG with R
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../emgr-part1/" title="Mining EMG with R part 1" class="md-nav__link">
      Mining EMG with R part 1
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Mining EMG with R part 2A
      </label>
    
    <a href="./" title="Mining EMG with R part 2A" class="md-nav__link md-nav__link--active">
      Mining EMG with R part 2A
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#gut-microbiota-disturbance-during-antibiotic-therapy-a-multi-omic-approach" title="Gut microbiota disturbance during antibiotic therapy: a multi-omic approach" class="md-nav__link">
    Gut microbiota disturbance during antibiotic therapy: a multi-omic approach
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gather-the-necessary-analysis-results-file-from-emg-portal" title="Gather the necessary analysis results file from EMG portal" class="md-nav__link">
    Gather the necessary analysis results file from EMG portal
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#1-taxonomy-who-is-there-and-in-what-proportion" title="1. Taxonomy: who is there and in what proportion?" class="md-nav__link">
    1. Taxonomy: who is there and in what proportion?
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-function-who-does-what-in-what-relative-proportion" title="2. Function: who does what, in what relative proportion?" class="md-nav__link">
    2. Function: who does what, in what relative proportion?
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../emgr-part2B/" title="Mining EMG with R part 2B" class="md-nav__link">
      Mining EMG with R part 2B
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#gut-microbiota-disturbance-during-antibiotic-therapy-a-multi-omic-approach" title="Gut microbiota disturbance during antibiotic therapy: a multi-omic approach" class="md-nav__link">
    Gut microbiota disturbance during antibiotic therapy: a multi-omic approach
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gather-the-necessary-analysis-results-file-from-emg-portal" title="Gather the necessary analysis results file from EMG portal" class="md-nav__link">
    Gather the necessary analysis results file from EMG portal
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#1-taxonomy-who-is-there-and-in-what-proportion" title="1. Taxonomy: who is there and in what proportion?" class="md-nav__link">
    1. Taxonomy: who is there and in what proportion?
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-function-who-does-what-in-what-relative-proportion" title="2. Function: who does what, in what relative proportion?" class="md-nav__link">
    2. Function: who does what, in what relative proportion?
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="part-2-functional-analysis-of-emg-genomic-and-transcriptomic-data-using-r">Part 2: Functional analysis of EMG genomic and transcriptomic data using R<a class="headerlink" href="#part-2-functional-analysis-of-emg-genomic-and-transcriptomic-data-using-r" title="Permanent link">&para;</a></h1>
<hr />
<h2 id="gut-microbiota-disturbance-during-antibiotic-therapy-a-multi-omic-approach">Gut microbiota disturbance during antibiotic therapy: a multi-omic approach<a class="headerlink" href="#gut-microbiota-disturbance-during-antibiotic-therapy-a-multi-omic-approach" title="Permanent link">&para;</a></h2>
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>Antibiotic usage strongly affects microbial intestinal metabolism and thereby impacts human health. Understanding this process and the underlying mechanisms remains a major research goal. Accordingly, we conducted the first comparative omic investigation of gut microbial communities in faecal samples taken at multiple time points from an individual subjected to Beta-lactam therapy.
<em>Perez-Cobas AE, et al. 2013 Gut 62(11), 1591-1601</em></p>
<p>We will explore the data generated by the study above. The data is available at:
<a href="https://www.ebi.ac.uk/metagenomics/projects/ERP001506">https://www.ebi.ac.uk/metagenomics/projects/ERP001506</a></p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>stringr<span class="p">)))</span>      <span class="c1"># String format</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>ade4<span class="p">)))</span>         <span class="c1"># Multivariate analysis</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>ggplot2<span class="p">)))</span>      <span class="c1"># Fancy plotting</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>grid<span class="p">)))</span>         <span class="c1"># Has the viewport function</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>dplyr<span class="p">)))</span>        <span class="c1"># Data manipulation</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>tidyr<span class="p">)))</span>        <span class="c1"># Data manipulation</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>FactoMineR<span class="p">)))</span>   <span class="c1"># Multivariate analysis</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>factoextra<span class="p">)))</span>   <span class="c1"># Visualize result of multivariate analysis</span>
</pre></div>
</td></tr></table>

<h3 id="gather-the-necessary-analysis-results-file-from-emg-portal">Gather the necessary analysis results file from EMG portal<a class="headerlink" href="#gather-the-necessary-analysis-results-file-from-emg-portal" title="Permanent link">&para;</a></h3>
<p>Read in the different contigency tables.
R provides a very versatile data structure that allows us to store various data types, namely the list.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>analysis_files <span class="o">&lt;-</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;BP_GO_abundances&quot;</span><span class="p">,</span><span class="s">&quot;BP_GO-slim_abundances&quot;</span><span class="p">,</span><span class="s">&quot;CC_GO_abundances&quot;</span><span class="p">,</span><span class="s">&quot;CC_GO-slim_abundances&quot;</span><span class="p">,</span><span class="s">&quot;GO_abundances&quot;</span><span class="p">,</span><span class="s">&quot;GO-slim_abundances&quot;</span><span class="p">,</span><span class="s">&quot;IPR_abundances&quot;</span><span class="p">,</span><span class="s">&quot;MF_GO_abundances&quot;</span><span class="p">,</span><span class="s">&quot;MF_GO-slim_abundances&quot;</span><span class="p">,</span><span class="s">&quot;phylum_taxonomy_abundances&quot;</span><span class="p">,</span><span class="s">&quot;taxonomy_abundances&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Read in all the contingency tables in a list making use of the lapply function</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>How can we access elements in the list?</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">attributes</span><span class="p">(</span>results<span class="p">)</span>
</pre></div>
</td></tr></table>
This approach enables us to assign a name to each entry in the list for unambiguous access to different contingency tables
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>colheader <span class="o">&lt;-</span> <span class="kp">gsub</span><span class="p">(</span>analysis_files<span class="p">,</span> pattern<span class="o">=</span><span class="s">&quot;-&quot;</span><span class="p">,</span> replacement <span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">)</span>
<span class="kp">names</span><span class="p">(</span>results<span class="p">)</span> <span class="o">&lt;-</span> colheader
<span class="kp">attributes</span><span class="p">(</span>results<span class="p">)</span>
</pre></div>
</td></tr></table>
Let&rsquo;s have a look at results object</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>str<span class="p">(</span>results<span class="p">,</span> list.len<span class="o">=</span><span class="m">4</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Each entry in the list can now be accessed by name. Give each row an explicit name before manipulating the table contents with dplyr functions. The main advantage of tbl_df (return a dataframe from a dataframe) is that it only displays a few rows and all the columns fit on the current screen, with the rest of the data displayed as text. We will be making extensive use of the dplyr package, which allows operations to be chained (&ldquo;%&gt;%&rdquo;); it is analogous to the linux pipe command (|), see here for more details: <a href="https://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html">https://cran.rstudio.com/web/packages/dplyr/vignettes/introduction.html</a>  </p>
<h3 id="1-taxonomy-who-is-there-and-in-what-proportion">1. Taxonomy: who is there and in what proportion?<a class="headerlink" href="#1-taxonomy-who-is-there-and-in-what-proportion" title="Permanent link">&para;</a></h3>
<p>We can optimise the display of the first few columns or so&hellip;</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15
16
17
18
19
20
21
22
23</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>taxonomy <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>phylum_taxonomy_abundances<span class="p">)</span>
taxonomy <span class="o">&lt;-</span> taxonomy <span class="o">%&gt;%</span> mutate<span class="p">(</span>phylum<span class="o">=</span><span class="kp">as.character</span><span class="p">(</span>phylum<span class="p">))</span>
tax_g <span class="o">&lt;-</span> taxonomy <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_G|um&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metagenomics&quot;</span><span class="p">)</span>
tax_t <span class="o">&lt;-</span> taxonomy <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_T|um&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metatranscriptomics&quot;</span><span class="p">)</span>
tax_a <span class="o">&lt;-</span> taxonomy <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_A|um&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Amplicon&quot;</span><span class="p">)</span>

renameSample <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>localdf<span class="p">){</span>
strippedcol <span class="o">&lt;-</span> <span class="kp">colnames</span><span class="p">(</span>localdf<span class="p">)</span>
<span class="kr">for</span> <span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="kp">nrow</span><span class="p">(</span>metadata<span class="p">)){</span>
strippedcol<span class="p">[</span><span class="kp">which</span><span class="p">(</span>strippedcol<span class="o">==</span><span class="kp">as.character</span><span class="p">(</span>metadata<span class="p">[</span>i<span class="p">,</span><span class="m">2</span><span class="p">]))]</span> <span class="o">&lt;-</span> <span class="kp">as.character</span><span class="p">(</span>metadata<span class="p">[</span>i<span class="p">,</span><span class="m">1</span><span class="p">])</span>
<span class="p">}</span>
<span class="kr">return</span><span class="p">(</span>strippedcol<span class="p">)</span>
<span class="p">}</span>

<span class="kp">colnames</span><span class="p">(</span>tax_g<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>tax_g<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>tax_t<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>tax_t<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>tax_a<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>tax_a<span class="p">)</span>
</pre></div>
</td></tr></table>
Then transform a wide contingency table into a long one for the purposes of plotting.</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>tax_metag <span class="o">&lt;-</span> tax_g <span class="o">%&gt;%</span> gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>tax_g<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
tax_metat <span class="o">&lt;-</span> tax_t <span class="o">%&gt;%</span> gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>tax_t<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
</pre></div>
</td></tr></table>
We can merge both metatranscriptomic and metagenomic relative occurrences of operational taxonomic units (OTU) per condition to visualise the community dynamic during treatment.</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>genomic_transcriptomics <span class="o">&lt;-</span> <span class="kp">rbind</span><span class="p">(</span>tax_metag<span class="p">,</span> tax_metat<span class="p">)</span>
</pre></div>
</td></tr></table>
Bar plot of relative abundance of species
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>genomic_transcriptomics <span class="o">%&gt;%</span> arrange<span class="p">(</span>desc<span class="p">(</span>prop<span class="p">))</span> <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span> y<span class="o">=</span>prop<span class="p">,</span>fill<span class="o">=</span>phylum<span class="p">))</span> <span class="o">+</span>
geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">,</span> position<span class="o">=</span><span class="s">&#39;fill&#39;</span><span class="p">,</span> aes<span class="p">(</span>fill <span class="o">=</span> phylum<span class="p">))</span> <span class="o">+</span>
facet_grid<span class="p">(</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
ggtitle<span class="p">(</span><span class="s">&#39;Species abundance through treatment per datatype&#39;</span><span class="p">)</span> <span class="o">+</span>
ylab<span class="p">(</span><span class="s">&#39;proportion abundance&#39;</span><span class="p">)</span> <span class="o">+</span>
theme_light<span class="p">()</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span> <span class="o">+</span>
coord_flip<span class="p">()</span>
</pre></div>
</td></tr></table></p>
<p><img alt="Bar plot of relative abundance of species" src="../images/EMGR-1.png" /></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What conclusion would you draw from this distribution?</p>
</div>
<p>The heatmap is of absolute Species counts per datatypes</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>genomic_transcriptomics <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span>y<span class="o">=</span>phylum<span class="p">))</span><span class="o">+</span> facet_grid<span class="p">(</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
geom_tile<span class="p">(</span>aes<span class="p">(</span>fill<span class="o">=</span>count<span class="p">))</span> <span class="o">+</span> scale_fill_gradient<span class="p">(</span>low<span class="o">=</span><span class="s">&quot;white&quot;</span><span class="p">,</span> high<span class="o">=</span><span class="s">&quot;darkblue&quot;</span><span class="p">)</span> <span class="o">+</span>
xlab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> ylab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span>
</pre></div>
</td></tr></table>
<img alt="heatmap is of absolute Species counts per datatypes" src="../images/EMGR-2.png" /></p>
<h3 id="2-function-who-does-what-in-what-relative-proportion">2. Function: who does what, in what relative proportion?<a class="headerlink" href="#2-function-who-does-what-in-what-relative-proportion" title="Permanent link">&para;</a></h3>
<p>Recall the various functional annotations in the created list.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">attributes</span><span class="p">(</span>results<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>As an initial exploration, we will focus on a broad GO annotation (GO-slim)</p>
<p>** (A) GO Biological processes**: relative occurrence during treatment</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bp_slim <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>BP_GO_slim_abundances<span class="p">)</span>
bp_slim <span class="o">&lt;-</span> bp_slim <span class="o">%&gt;%</span> mutate<span class="p">(</span>description<span class="o">=</span><span class="kp">as.character</span><span class="p">(</span>description<span class="p">))</span>
bp_slim_g <span class="o">&lt;-</span> bp_slim <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_G|description&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metagenomics&quot;</span><span class="p">)</span>
bp_slim_t <span class="o">&lt;-</span> bp_slim <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_T|description&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metatranscriptomics&quot;</span><span class="p">)</span>
bp_slim_a <span class="o">&lt;-</span> bp_slim <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_A|description&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Amplicon&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Give meaningful names to conditions, as for the taxonomic case above.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">colnames</span><span class="p">(</span>bp_slim_g<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>bp_slim_g<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>bp_slim_t<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>bp_slim_t<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>bp_slim_a<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>bp_slim_a<span class="p">)</span>
bp_slim_metagenomics <span class="o">&lt;-</span> bp_slim_g <span class="o">%&gt;%</span>
gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>bp_slim_g<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>

bp_slim_metatransciptomics <span class="o">&lt;-</span> bp_slim_t <span class="o">%&gt;%</span>
gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>bp_slim_t<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
</pre></div>
</td></tr></table>

<p>Merge both metatranscriptomic and metagenomic relative occurences of biological process terms per condition to visualise the relative abundance of biological process terms that are dynamic during treatment.</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bp_genomic_transcriptomics <span class="o">&lt;-</span> <span class="kp">rbind</span><span class="p">(</span>bp_slim_metagenomics<span class="p">,</span> bp_slim_metatransciptomics<span class="p">)</span>
</pre></div>
</td></tr></table>
Bar plot of relative abundance of Biological Processes</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bp_genomic_transcriptomics <span class="o">%&gt;%</span> arrange<span class="p">(</span>desc<span class="p">(</span>prop<span class="p">))</span> <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span> y<span class="o">=</span>prop<span class="p">,</span>fill<span class="o">=</span>description<span class="p">))</span> <span class="o">+</span>
geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">,</span> position<span class="o">=</span><span class="s">&#39;fill&#39;</span><span class="p">,</span> aes<span class="p">(</span>fill <span class="o">=</span> description<span class="p">))</span> <span class="o">+</span>
facet_wrap<span class="p">(</span><span class="o">~</span>DataType<span class="p">,</span> ncol<span class="o">=</span><span class="m">1</span><span class="p">)</span> <span class="o">+</span>
ggtitle<span class="p">(</span><span class="s">&#39;Biological processess relative abundance through \n treatment per datatype&#39;</span><span class="p">)</span> <span class="o">+</span>
ylab<span class="p">(</span><span class="s">&#39;proportion abundance&#39;</span><span class="p">)</span> <span class="o">+</span>
theme_light<span class="p">()</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span> <span class="o">+</span> coord_flip<span class="p">()</span> <span class="o">+</span>
theme<span class="p">(</span>legend.position<span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p><img alt="Bar plot of relative abundance of Biological Processes" src="../images/EMGR-3.png" /></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>What can you conclude from this plot?</p>
<p>Heat map of absolute biological process counts per datatype</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bp_genomic_transcriptomics <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span>y<span class="o">=</span>description<span class="p">))</span><span class="o">+</span> facet_grid<span class="p">(</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
geom_tile<span class="p">(</span>aes<span class="p">(</span>fill<span class="o">=</span>count<span class="p">))</span> <span class="o">+</span> scale_fill_gradient<span class="p">(</span>low<span class="o">=</span><span class="s">&quot;white&quot;</span><span class="p">,</span> high<span class="o">=</span><span class="s">&quot;darkblue&quot;</span><span class="p">)</span> <span class="o">+</span>
xlab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> ylab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span>
</pre></div>
</td></tr></table>

<p><img alt="Heat map of absolute biological process counts per datatype" src="../images/EMGR-4.png" /></p>
<p>Heat map of biological process proportions per datatype</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bp_genomic_transcriptomics <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span>y<span class="o">=</span>description<span class="p">))</span><span class="o">+</span> facet_grid<span class="p">(</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
geom_tile<span class="p">(</span>aes<span class="p">(</span>fill<span class="o">=</span>prop<span class="p">))</span> <span class="o">+</span> scale_fill_gradient<span class="p">(</span>low<span class="o">=</span><span class="s">&quot;white&quot;</span><span class="p">,</span> high<span class="o">=</span><span class="s">&quot;darkblue&quot;</span><span class="p">)</span> <span class="o">+</span>
xlab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> ylab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span>
</pre></div>
</td></tr></table>
<img alt="Heat map of biological process proportions per datatype" src="../images/EMGR-5.png" /></p>
<p><strong>(B) GO Cellular Compartment</strong>: relative occurence during treatment</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cc_slim <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>CC_GO_slim_abundances<span class="p">)</span>
cc_slim <span class="o">&lt;-</span> cc_slim <span class="o">%&gt;%</span> mutate<span class="p">(</span>description<span class="o">=</span><span class="kp">as.character</span><span class="p">(</span>description<span class="p">))</span>
cc_slim_g <span class="o">&lt;-</span> cc_slim <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_G|description&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metagenomics&quot;</span><span class="p">)</span>
cc_slim_t <span class="o">&lt;-</span> cc_slim <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_T|description&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metatranscriptomics&quot;</span><span class="p">)</span>
cc_slim_a <span class="o">&lt;-</span> cc_slim <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_A|description&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Amplicon&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>
Give meaningful names to conditions as for the taxonomic case above</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">colnames</span><span class="p">(</span>cc_slim_g<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>cc_slim_g<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>cc_slim_t<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>cc_slim_t<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>cc_slim_a<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>cc_slim_a<span class="p">)</span>
</pre></div>
</td></tr></table>
Calculate the relative occurrence of cellular compartments per day of treatment for both metagenomic and transcriptomic data types.
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8
9</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cc_slim_metagenomics <span class="o">&lt;-</span> cc_slim_g <span class="o">%&gt;%</span>
gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>cc_slim_g<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>

cc_slim_metatransciptomics <span class="o">&lt;-</span> cc_slim_t <span class="o">%&gt;%</span>
gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>cc_slim_t<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
</pre></div>
</td></tr></table>
Merge both metatranscriptomic and metagenomic relative occurrences of biological processes per condition to visualise the relative abundance of biological process terms that are dynamic during treatment.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cc_genomic_transcriptomics <span class="o">&lt;-</span> <span class="kp">rbind</span><span class="p">(</span>cc_slim_metagenomics<span class="p">,</span> cc_slim_metatransciptomics<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Bar plot of relative abundances of cellular compartment</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8
9</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cc_genomic_transcriptomics <span class="o">%&gt;%</span> arrange<span class="p">(</span>desc<span class="p">(</span>prop<span class="p">))</span> <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span> y<span class="o">=</span>prop<span class="p">,</span>fill<span class="o">=</span>description<span class="p">))</span> <span class="o">+</span>
geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">,</span> position<span class="o">=</span><span class="s">&#39;fill&#39;</span><span class="p">,</span> aes<span class="p">(</span>fill <span class="o">=</span> description<span class="p">))</span> <span class="o">+</span>
facet_wrap<span class="p">(</span><span class="o">~</span>DataType<span class="p">,</span> ncol<span class="o">=</span><span class="m">1</span><span class="p">)</span> <span class="o">+</span>
ggtitle<span class="p">(</span><span class="s">&#39;Cellular compartment relative abundance through \n</span>
<span class="s">treatment per datatype&#39;</span><span class="p">)</span> <span class="o">+</span>
ylab<span class="p">(</span><span class="s">&#39;proportion abundance&#39;</span><span class="p">)</span> <span class="o">+</span> theme_light<span class="p">()</span> <span class="o">+</span>
theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span> <span class="o">+</span>
coord_flip<span class="p">()</span> <span class="o">+</span> theme<span class="p">(</span>legend.position<span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p><img alt="Bar plot of relative abundances of cellular compartment" src="../images/EMGR-6.png" /></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What can you conclude from this plot?</p>
</div>
<p>Heat map of absolute cellular compartment term counts per datatype</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cc_genomic_transcriptomics <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span>y<span class="o">=</span>description<span class="p">))</span><span class="o">+</span> facet_grid<span class="p">(</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
geom_tile<span class="p">(</span>aes<span class="p">(</span>fill<span class="o">=</span>count<span class="p">))</span> <span class="o">+</span> scale_fill_gradient<span class="p">(</span>low<span class="o">=</span><span class="s">&quot;white&quot;</span><span class="p">,</span> high<span class="o">=</span><span class="s">&quot;darkblue&quot;</span><span class="p">)</span> <span class="o">+</span>
xlab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> ylab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span>
</pre></div>
</td></tr></table>

<p><img alt="Heat map of absolute cellular compartment term counts per datatype" src="../images/EMGR-7.png" /></p>
<p>Heat map of Cellular compartment term proportions per datatype</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cc_genomic_transcriptomics <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span>y<span class="o">=</span>description<span class="p">))</span><span class="o">+</span> facet_grid<span class="p">(</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
geom_tile<span class="p">(</span>aes<span class="p">(</span>fill<span class="o">=</span>prop<span class="p">))</span> <span class="o">+</span> scale_fill_gradient<span class="p">(</span>low<span class="o">=</span><span class="s">&quot;white&quot;</span><span class="p">,</span> high<span class="o">=</span><span class="s">&quot;darkblue&quot;</span><span class="p">)</span> <span class="o">+</span>
xlab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> ylab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span>
</pre></div>
</td></tr></table>

<p><img alt="Heat map of Cellular compartment term proportions per datatype" src="../images/EMGR-8.png" /></p>
<p><strong>(C) GO Molecular function</strong>: relative occurence during treatment</p>
<p>Extract the data from the data structure &ldquo;results&rdquo; created earlier.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mf_slim <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>MF_GO_slim_abundances<span class="p">)</span>
mf_slim <span class="o">&lt;-</span> mf_slim <span class="o">%&gt;%</span> mutate<span class="p">(</span>description<span class="o">=</span><span class="kp">as.character</span><span class="p">(</span>description<span class="p">))</span>
mf_slim_g <span class="o">&lt;-</span> mf_slim <span class="o">%&gt;%</span> 
            select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_G|description&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span> 
            mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metagenomics&quot;</span><span class="p">)</span>
mf_slim_t <span class="o">&lt;-</span> mf_slim <span class="o">%&gt;%</span> 
            select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_T|description&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span> 
            mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metatranscriptomics&quot;</span><span class="p">)</span> 
mf_slim_a <span class="o">&lt;-</span> mf_slim <span class="o">%&gt;%</span> 
            select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_A|description&quot;</span><span class="p">))</span>  <span class="o">%&gt;%</span> 
            mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Amplicon&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Give meaningful names to conditions as for the taxonomic case above</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">colnames</span><span class="p">(</span>mf_slim_g<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>mf_slim_g<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>mf_slim_t<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>mf_slim_t<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>mf_slim_a<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>mf_slim_a<span class="p">)</span>

mf_slim_metagenomics <span class="o">&lt;-</span> mf_slim_g <span class="o">%&gt;%</span> 
                        gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>mf_slim_g<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
                        group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span> 
                        mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>

mf_slim_metatransciptomics <span class="o">&lt;-</span> mf_slim_t <span class="o">%&gt;%</span> 
                              gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>mf_slim_t<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
                              group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
                              mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
</pre></div>
</td></tr></table>

<p>Merge both metatranscriptomic and metagenomic relative occurrences of biological processes per condition to visualise the relative abundances of biological process terms that are dynamic during treatment.  </p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mf_genomic_transcriptomics <span class="o">&lt;-</span> <span class="kp">rbind</span><span class="p">(</span>mf_slim_metagenomics<span class="p">,</span> mf_slim_metatransciptomics<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Barplot of relative abundance Molecular function</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mf_genomic_transcriptomics <span class="o">%&gt;%</span> arrange<span class="p">(</span>desc<span class="p">(</span>prop<span class="p">))</span> <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span> y<span class="o">=</span>prop<span class="p">,</span>fill<span class="o">=</span>description<span class="p">))</span> <span class="o">+</span>
geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">,</span> position<span class="o">=</span><span class="s">&#39;fill&#39;</span><span class="p">,</span> aes<span class="p">(</span>fill <span class="o">=</span> description<span class="p">))</span> <span class="o">+</span>
facet_wrap<span class="p">(</span><span class="o">~</span>DataType<span class="p">,</span> ncol<span class="o">=</span><span class="m">1</span><span class="p">)</span> <span class="o">+</span>
ggtitle<span class="p">(</span><span class="s">&#39;Molecular function relative abundance through</span>
<span class="s">\n treatment per datatype&#39;</span><span class="p">)</span> <span class="o">+</span>
ylab<span class="p">(</span><span class="s">&#39;proportion abundance&#39;</span><span class="p">)</span> <span class="o">+</span>
theme_light<span class="p">()</span> <span class="o">+</span>
theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span> <span class="o">+</span>
coord_flip<span class="p">()</span> <span class="o">+</span>
theme<span class="p">(</span>legend.position<span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span><span class="c1"># + guides(fill=guide_legend(ncol=2))</span>
</pre></div>
</td></tr></table>
Heatmap of Absolute Molecular Function term counts through treatment</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mf_genomic_transcriptomics <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span>y<span class="o">=</span>description<span class="p">))</span> <span class="o">+</span> facet_grid<span class="p">(</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
geom_tile<span class="p">(</span>aes<span class="p">(</span>fill<span class="o">=</span>count<span class="p">))</span> <span class="o">+</span> scale_fill_gradient<span class="p">(</span>low<span class="o">=</span><span class="s">&quot;white&quot;</span><span class="p">,</span> high<span class="o">=</span><span class="s">&quot;darkblue&quot;</span><span class="p">)</span> <span class="o">+</span>
xlab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> ylab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span>
</pre></div>
</td></tr></table>
Heat map of relative occurrence of molecular function term counts throughout treatment</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What can you conclude from the plots?</p>
</div>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../emgr-part1/" title="Mining EMG with R part 1" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Mining EMG with R part 1
              </span>
            </div>
          </a>
        
        
          <a href="../emgr-part2B/" title="Mining EMG with R part 2B" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Mining EMG with R part 2B
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-c35428f87f.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>